WebRoot/h/changepass

```diff
@@ -80,63 +83,279 @@
+                                <div class="signInAndLabel">
+                                    <div>
+                                        <input type="submit" class="loginButton" value="<fmt:message key="save"/>" />
+                                    </div>
+                                </div>
+                            </div>
+                        </div>
+                    </c:if>
+                </form>
             </div>
             <div class="decor1"></div>
         </div>
-    </div> 
+    </div>
+    <script>
+        function getElement(id) {
+            return document.getElementById(id);
+        }
+        function check(checkImg, closeImg) {
+            closeImg.style.display = "none";
+            checkImg.style.display = "inline";
+        }
+        function unCheck(checkImg, closeImg) {
+            closeImg.style.display = "inline";
+            checkImg.style.display = "none";
+        }
+        function resetImg(condition, checkImg, closeImg){
+            condition ? check(checkImg, closeImg) : unCheck(checkImg, closeImg);
+        }
+        function showPassword(x, y, z) {
+            if (x.type === "password") {
+                x.type = "text";   
+                y.style.display = "none";
+                z.style.display = "block";
+            } else {
+                x.type = "password";
+                y.style.display = "block";
+                z.style.display = "none";
+            }
+        }
+        function showOldPassword() {
+            var x = getElement("oldPassword");
+            var y = getElement("oldPasswordShowSpan");
+            var z = getElement("oldPasswordHideSpan");
+            this.showPassword(x, y, z);
+        }
+        function showNewPassword() {
+            var x = getElement("newPassword");
+            var y = getElement("newPasswordShowSpan");
+            var z = getElement("newPasswordHideSpan");
+            this.showPassword(x, y, z);
+        }
+        function showConfirmPassword() {
+            var x = getElement("confirm");
+            var y = getElement("confirmShowSpan");
+            var z = getElement("confirmHideSpan");
+            this.showPassword(x, y, z);
+        }
+
+        var newPasswordInput = getElement("newPassword");
+
+        // Function to check special character
+        function isAsciiPunc(ch) {
+            return (ch >= 33 && ch <= 47) || // ! " # $ % & ' ( ) * + , - . /
+            (ch >= 58 && ch <= 64) || // : ; < = > ? @
+            (ch >= 91 && ch <= 96) || // [ \ ] ^ _ `
+            (ch >= 123 && ch <= 126); // { | } ~
+        }
+
+        function parseCharsFromPassword(passwordString) {
+            const uppers = [],
+                lowers = [],
+                numbers = [],
+                punctuations = [],
+                invalidChars = [],
+                invalidPuncs = [];
+
+            const chars = passwordString.split('');
+
+            chars.forEach(char => {
```

Comment:
not sure all browsers will be supporting arrow functions

Reply:
Fixed.